tests/integ/scala/test_dataframe_join_suite.py

```diff
@@ -1667,3 +1667,41 @@ def aggregate(input):
         Utils.check_answer(aggregate(df), [Row("James", 3000)])
     finally:
         session._join_alias_fix = origin
+
+
+@pytest.mark.parametrize("how", ["inner", "left", "cross", None])
+@pytest.mark.parametrize("with_condition", [True, False])
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="Lateral join is not supported in Local Testing",
+)
+def test_lateral_join_behaviors(session, how, with_condition):
+    df1 = session.create_dataframe([[1, 2], [3, 4], [5, 6]], schema=["a", "b"])
+    df2 = session.create_dataframe([[1, 7], [3, 8]], schema=["a", "c"])
+
+    condition = df1.a == df2.a if with_condition else None
+    if how is None:
+        df = df1.lateral_join(df2, condition)
+    else:
+        df = df1.lateral_join(df2, condition, how=how)
```

Comment:
nit: this can be simplified to one line since how defaults to None
```suggestion
    df = df1.lateral_join(df2, condition, how=how)
```